twoliter/embedded/ocihelper

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+digest_from_blob() {
+  local -n bytes
+  bytes="${1:?}"
+  echo "${bytes}" | sha256sum | awk '{ print $1 };'
+}
+
+digest_from_file() {
+  local -n file_path
+  file_path="${1:?}"
+  sha256sum < "${file_path}" | awk '{ print $1 };'
+}
+
+# Create a layer for a configuration
+# create_json_layer $OUT_DIR $JSON_BLOB
+create_json_layer() {
+  local -n out_dir
+  local -n json_blob
+  local digest
+
+  out_dir="${1:?}"
+  json_blob="${2:?}"
+  digest=$(digest_from_blob "${json_blob}")
+
+  echo "${json_blob}" > "${out_dir}/blobs/sha256/${digest}"
+  echo "${digest}"
+}
+
+get_size() {
+  local -n file_path
+
+  file_path="${1:?}"
+  # Use xargs to strip leading and trailing whitespace
+  wc -c < "${file_path}" | xargs
```

Comment:
this would be more idiomatic - `xargs` is not really meant for whitespace removal:
```suggestion
  wc -c < "${file_path}" | tr -d '[:space:]'
```

But I would use `stat -c %s ...` instead, and you don't really need to remove the newline if you're assigning the result to a variable.